libs/payment/api/src/lib/payment-provider/mollie-payment-provider.ts

```diff
@@ -0,0 +1,270 @@
+import {
+  BasePaymentProvider,
+  CheckIntentProps,
+  CreatePaymentIntentProps,
+  Intent,
+  IntentState,
+  InvoiceWithItems,
+  PaymentProviderProps,
+  WebhookForPaymentIntentProps,
+  WebhookResponse
+} from './payment-provider'
+import {logger} from '@wepublish/utils/api'
+import {PaymentState} from '@prisma/client'
+import createMollieClient, {
+  MollieClient,
+  SequenceType,
+  Payment,
+  PaymentMethod
+} from '@mollie/api-client'
+import MaybeArray from '@mollie/api-client/dist/types/types/MaybeArray'
+
+export interface MolliePaymentProviderProps extends PaymentProviderProps {
+  apiKey: string
+  webhookEndpointSecret: string
+  apiBaseUrl: string
+  methods?: string[]
+}
+
+const erroredPaymentIntent = {
+  intentID: 'error',
+  intentSecret: '',
+  intentData: 'error',
+  state: PaymentState.requiresUserAction,
+  errorCode: 'error'
+}
+
+type MolliePaymentMetadata = {
+  paymentID?: string
+  mail?: string
+}
+
+function mapMollieEventToPaymentStatus(event: string): PaymentState {
+  switch (event) {
+    case 'failed':
+    case 'expired':
+      return PaymentState.requiresUserAction
+    case 'open':
+    case 'authorized':
+    case 'pending':
+      return PaymentState.processing
+    case 'paid':
+      return PaymentState.paid
+    case 'canceled':
+      return PaymentState.canceled
+    default:
+      throw new Error(`Unhandled ${event} state received from mollie`)
+  }
+}
+
+function calculateAndFormatAmount(invoice: InvoiceWithItems) {
+  return `${(
+    invoice.items.reduce(
+      (prevItem, currentItem) => prevItem + currentItem.amount * currentItem.quantity,
+      0
+    ) / 100
+  ).toFixed(2)}`
```

Comment:
\`${}\` should not be needed here, because toFixed already returns a String

Reply:
True :-)